CONTRIBUTING.md

```diff
@@ -627,6 +631,74 @@ required.
 Essentially, the steps you have to take to deploy manually are the same that
 `skaffold` will perform, and they're outlined at the previous section.
 
+#### Kubernetes E2E tests
+
+Kubernetes integration is quite complicated, a lot of things can go wrong.
+
+To cope with the complexity and ensure we maintain high quality, we use
+E2E (end-to-end) tests.
+
+> E2E tests normally run at CI, so there's typically no need to run them
+> manually.
+
+##### Requirements
+
+* `kubernetes` cluster (`minikube` has special support, but any cluster should
+  work)
+* `docker`
+* `kubectl`
+* `bash`
+
+Vector release artifacts are prepared for E2E tests, so the ability to do that
+is required too, see Vector [docs](https://vector.dev) for more details.
+
+##### Running the E2E tests
+
+To run the E2E tests, use the following command:
+
+```shell
+CONTAINER_IMAGE_REPO=<your name>/vector-test make test-e2e-kubernetes
+```
+
+Where `CONTAINER_IMAGE_REPO` is the docker image repo name to use, without part
+after the `:`. Replace `<your name>` with your Docker Hub username.
+
+This will run Kubernetes E2E tests suite. If the command exit with 0 status
+code - tests passed.
+
+You can also pass additional parameters to adjust the behavior of the test:
+
+* `QUICK_BUILD=true` - use development build and a skaffold image from the dev
```

Comment:
How about `DEVELOPMENT=true`? I'd like to use that same env in other tasks so it'd be nice to keep the number of vars down.

Reply:
I'm pretty sure `DEVELOPMENT` didn't exist when I implemented this :smile: I'll see it if fits - we might want to deliberately have those different. I'll check and respond here.

Reply:
It doesn't yet! :( You've probably noticed we use separate make jobs for this distinction. Maybe that's appropriate here too?

Reply:
You're thinking about `make test-e2e-kubernetes` and a `make test-e2e-kubernetes-quick`, sth like that? I thought about doing it like that, but I discarded it because we'd still need to pass env vars is some cases, and adding entry points complicates things worse than just having one configurable one imo. It's a good idea though, thx for bringing this up.

I understand what you mean with `DEVELOPMENT` now. Well, I wouldn't worry about it now. Let's streamline it when it becomes a problem. For now, `QUICK_BUILD` captures the distinction with the regular flow better (for the "how do I make it quick!!1" situations :smile: ), and with lack of widely used `DEVELOPMENT` var - I think it'd be easier to find.

Reply:
I do not think `QUICK_BUILD` is a good name for this value.